### ISSUE TITLE
Should competition radius be different for seedlings and adults?

### ISSUE THREAD

**Issue by hrlai** [wengngai/NSSF_IBMs#12]
If I understand correctly, in many functions seedlings and adults have the same neighbourhood radii. But shouldn't smaller seedlings interact at a smaller radius? To complicate matters, a seedling probably has a larger response radius to adult competitors and another smaller response radius to seedling competitors...

**Comment by wengngai** [comment]
I think there might be two related but separate issues here: (i) distance attenuation of competitive effect (ii) magnitude/strength of effect/response. But the limitations in the way competition was measured in the field in the original data makes it very hard to distinguish the two..
I don't really think there's a better way around this because we simply have to model the effect exactly as it was measured and thus modelled in the vital rate model, and there were not many options available.
But I'm hoping that the competition coefficient (which rightly is a measure of magnitude/strength of the competitive effect) may in some ways also capture the effect of the greater distance attenuation in some life stages as well, since this would just result in an overall smaller coefficient value. I hope I'm making sense haha..

**Comment by hrlai** [comment]
Yes what you said makes lots of sense! It is hard to be sure of the competition equations from the codes, but yes the competitive coefficient should soak up the distance effects too (in fact a reparameterisation of the competitive coefs will give per neighbour per distance unit).

Speaking of these coefficients, did you constrain their signs to always be competitive? I noticed that some can be positive and some negative. Without knowing the link-function etc. it is hard to tell whether they are always competitive. I am just worried that if there is facilitation, then we may never get the two species to coexist (though you may not be aiming for it), and Prunus will just dominate the landscape eventually.

**Comment by wengngai** [comment]
I constrained the coefficients in brms models (i.e. survival, adult growth) to be negative (competitive). this was mainly because of the lower bound = 0 condition that many of the parameters took (which translated to certain constrains in their submodel predictor coefficients). However, it was possible, when the coefficient was small enough, for the random slope to overwhelm this effect, so that the net response of a species to competition became positive. But this was quite rare.
For seedling growth, the coefficients were not constrained. 
Although I agree that facilitation definitely happens here or there, our measure of competition is identity-less (sums up the effect of all neighbour species), but I'm thinking that if there's facilitation, the interaction would probably be a bit more pairwise specific? eg, sp A --> sp B, sp A --> sp C, sp C --> sp D, but not any other direction or combinations. If facilitation is smaller in magnitude and specific to only a subset of interactions, then when we measures the summed effect of neighbours it would be drowned out by the competitive effect. This is both good (in that it simplifies things for us, and probably means our contraining of the coefficient was not wrong) and bad (we're missing out a phenomenon in the model, which could potentially make a big dfiference in prediction outcomes). I'm not very sure what we can/should do in this regard..
